core/src/test/java/org/commonjava/maven/ext/core/fixture/TestUtils.java

```diff
@@ -90,24 +91,38 @@ public static Model getDummyModel ()
         return result;
     }
 
+    // Public API for manipulator-examples-project
+    public static ManipulationSession createSession( Properties p, File pom ) throws ManipulationException
+    {
+        return createSessionAndManager( p, pom ).getSession();
+    }
+
     // Public API for manipulator-examples-project
     public static ManipulationSession createSession( Properties p ) throws ManipulationException
     {
-        return createSessionAndManager( p ).getSession();
+        return createSessionAndManager( p, null ).getSession();
     }
 
     // Public API for manipulator-examples-project
     @SuppressWarnings( "deprecation" )
-    public static SMContainer createSessionAndManager( Properties p ) throws ManipulationException
+    public static SMContainer createSessionAndManager( Properties  p ) throws ManipulationException
+    {
+        return createSessionAndManager( p, null );
+    }
+
+    // Public API for manipulator-examples-project
+    @SuppressWarnings( "deprecation" )
+    public static SMContainer createSessionAndManager( Properties p, File pom ) throws ManipulationException
```

Comment:
Can we add Javadoc for this and the preceding method

Reply:
There's no javadoc on most `public` methods here. Should we add it?

I still don't know if this is the right API. In some cases, we create a session without a POM. For example, a `ModelIO` requires a session and is generally used for resolving files outside of a project. The old version of the code seemed to handle the case of not having a file.

Reply:
I suppose we _should_ add it ; we haven't been especially diligent

Reply:
Agreed to leave it for now as its test code.